domain-model-assistant/Assets/Components/Scripts/Diagram.cs

```diff
@@ -128,53 +123,22 @@ void Update()
             (_currentMode == CanvasMode.AddingClass && InputExtender.MouseExtender.IsSingleClick()))
         {
             _updateNeeded = true;
-            AddClass("Class" + (compartmentedRectangles.Count + 1), Input.mousePosition);
+            WebCore.AddClass("Class" + (compartmentedRectangles.Count + 1), Input.mousePosition);
```

Comment:
Now we call WebCORE to do operations on the class diagram, which simplifies the `Diagram` class significantly!